component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpBasicTests.java

```diff
@@ -127,116 +139,125 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                "@info(name = 'queryB') "
+                        + "from inputStreamB "
+                        + "select *  "
+                        + "insert into outputStreamB;"
+                        );
         ExecutionPlanRuntime executionPlanRuntime = siddhiManager
-                .createExecutionPlanRuntime(inStreamDefinition + queryA);
-        executionPlanRuntime.addCallback("queryA", new QueryCallback() {
-            @Override
-            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
-                EventPrinter.print(timeStamp, inEvents, removeEvents);
-                for (Event event : inEvents) {
-                    receivedEventNameListA.add(event.getData(0).toString());
-                }
-            }
-        });
+                .createExecutionPlanRuntime(inStreamDefinitionA +  queryA);
         try {
+            executionPlanRuntime.addCallback("queryA", new QueryCallback() {
+                @Override
+                public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                    EventPrinter.print(timeStamp, inEvents, removeEvents);
+                    for (Event event : inEvents) {
+                        receivedEventNameListA.add(event.getData(0).toString());
+                    }
+                }
+            });
             executionPlanRuntime = siddhiManager
-                    .createExecutionPlanRuntime(inStreamDefinition2 + queryB);
+                    .createExecutionPlanRuntime(inStreamDefinitionB + queryB);
+            executionPlanRuntime.addCallback("queryB", new QueryCallback() {
+                @Override
+                public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                    EventPrinter.print(timeStamp, inEvents, removeEvents);
+                    for (Event event : inEvents) {
+                        receivedEventNameListB.add(event.getData(0).toString());
+                    }
+                }
+            });
+            //To check weather only one is deployed
+            Assert.assertEquals(1, executionPlanRuntime.getSources().size());
         } catch (Exception e) {
-            Assert.assertEquals("Listener URL http://localhost:9005/endpoints/" +
-                    "RecPro already connected in StreamDefinition{id='inputStreamB', attributeList" +
-                    "=[Attribute{id='name', type=STRING}, Attribute{id='age', type=INT}, Attribute{id='country', " +
-                    "type=STRING}], annotations=[Annotation{name='source', elements=[Element{key='type', value='http'}"
-                    + ", Element{key='receiver.url', value='http://localhost:9005/endpoints/RecPro'}, Element{key=" +
-                    "'basic.auth.enabled', value='false'}], annotations=[Annotation{name='map', " +
-                    "elements=[Element{key='type', value='xml'}], annotations=[]}]}]}", e.getMessage());
+            throw new RuntimeException(e);
```

Comment:
What is the reason for doing this?

Reply:
Fixed in 602e96bfa91ea786afe9f9bd414b1bc8b1787631